absl/base/internal/sysinfo.cc

```diff
@@ -76,6 +76,7 @@ static int GetNumCPUs() {
 #if defined(_WIN32)
 
 static double GetNominalCPUFrequency() {
+#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY != WINAPI_FAMILY_APP)
```

Comment:
I'm curious about the thinking here.  In the case where WINAPI_FAMILY is not defined, is it correct to assume registry access is not available?  Might this cause problems on older toolchains where this macro is not yet defined?

Is it possible that this wants to be
```
#if !defined(WINAPI_FAMILY) || (WINAPI_FAMILY != WINAPI_FAMILY_APP)
```
?  (This is not intended to be a leading question; I'm really asking.)

Reply:
Hi vslashg, the following was extracted from the winapifamily.h public header in the Windows SDK.

`WINAPI_FAMILY` might be undefined, or defined as:
* `WINAPI_FAMILY_PC_APP` - Windows Store Applications
* `WINAPI_FAMILY_PHONE_APP` - Windows Phone Applications
* `WINAPI_FAMILY_SYSTEM` - Windows Drivers and Tools
* `WINAPI_FAMILY_SERVER` - Windows Server Applications
* `WINAPI_FAMILY_DESKTOP_APP` - Windows Desktop Applications

If no WINAPI_FAMILY value is specified, then all APIs available to Windows desktop applications are exposed.
```c
#ifndef WINAPI_FAMILY
#define WINAPI_FAMILY WINAPI_FAMILY_DESKTOP_APP
#endif
```
So, not having WINAPI_FAMILY defined should work fine for code written for desktop apps.

Did you have any problem with this code in an old toolchain?

Reply:
BTW, `#include <winapifamily.h>` is literally the first line of Windows.h.

Reply:
Hi @vslashg , I went to abseil compatibility page (https://abseil.io/docs/cpp/platforms/platforms) and noticed that abseil supports down to "MVC 2015 Update 3, MSVC 2017", but no specific Windows version or SDK version is mentioned.
I think you might be on something here. I'm not sure if the SDK for Windows 7 has winapifamily.h.
Can you confirm what is the minimal Windows/SDK version abseil supports?

Reply:
In general, we support a version until 5 years after its replacement is available. In this case, I think we can probably ignore windows 7.
@vslashg WDYT?

Reply:
I've read that Google is going to support Win7 until at least July 15, 2021 (https://cloud.google.com/blog/products/chrome-enterprise/how-chrome-is-helping-enterprises-still-using-windows-7). That's why I've created a Win7 VM + VS 2008 to check my change.
What is not clear to me is which toolchain abseil uses on Win7. Abseil's website (https://abseil.io/docs/cpp/platforms/platforms) informs that MVC 2015 Update 3 is used, but it doesn't say which SDK version is used.